aesara/tensor/__init__.py

```diff
@@ -80,7 +80,7 @@ def get_vector_length(v: Any):
 
 
 @singledispatch
-def _get_vector_length(op: Union[Op, Variable], var: Variable) -> NoReturn:
+def _get_vector_length(op: Union[Op, Variable], var: Variable):
```

Comment:
Same here; why remove the type hint?

Reply:
Here I couldn't figure out the correct type, but it's not `NoReturn` on the dispatched functions (similar to above).

The `NoReturn` was added in a PR that used monkeytype to infer types. Looks like it didn't do a great job for dispatch functions.